addons/account/models/account.py

```diff
@@ -253,6 +256,24 @@ def _check_currency(self):
             if self.default_debit_account_id and not self.default_debit_account_id.currency_id.id == self.currency_id.id:
                 raise UserError(_('Configuration error!\nThe currency of the journal should be the same than the default debit account.'))
 
+    @api.one
```

Comment:
The partner field actually is visible, it's the one labelled 'Account Holder'. When configuring an account from a journal, it defaults to journal.company_id.partner_id. And this field also has an useful onchange that fills in the address fields of the bank account using the partner data.

The current logic for the res.partner.bank is that the company_id field is for multicompany settings (meaning which company internal to odoo owns the record) and the partner_id field is the legal person that owns the account.

For this reason, I think that when configuring a bank account, the fields company_id and partner_id should be inferred from the journal company_id and be invisible to the user. Only the address fields should be editable.

In order to have an account that belongs to the company and is managed by a holding, I think the user would have to input the holding legal details in the address fields.